pkg/controller/controller_test.go

```diff
@@ -2486,16 +2494,24 @@ func assertServiceBindingOperationSuccess(t *testing.T, obj runtime.Object, oper
 
 func assertServiceBindingOperationSuccessWithParameters(t *testing.T, obj runtime.Object, operation v1beta1.ServiceBindingOperation, externalParameters map[string]interface{}, externalParametersChecksum string, originalBinding *v1beta1.ServiceBinding) {
 	var (
-		reason      string
-		readyStatus v1beta1.ConditionStatus
+		reason       string
+		readyStatus  v1beta1.ConditionStatus
+		unbindStatus v1beta1.ServiceBindingUnbindStatus
 	)
 	switch operation {
 	case v1beta1.ServiceBindingOperationBind:
 		reason = successInjectedBindResultReason
 		readyStatus = v1beta1.ConditionTrue
+		unbindStatus = v1beta1.ServiceBindingUnbindStatusRequired
 	case v1beta1.ServiceBindingOperationUnbind:
 		reason = successUnboundReason
 		readyStatus = v1beta1.ConditionFalse
+		switch originalBinding.Status.UnbindStatus {
```

Comment:
The UnbindStatus should always be succeeded after a successful unbind request.

Reply:
that is not true for the case of notrequired.

Reply:
If the UnbindStatus was Not required, then an unbind request should never have been sent.

Reply:
No, that is not true. This is for the request to the broker. Operation refers to an operation being performed on the broker. That is why operations are Bind and Unbind instead of Create and Delete. That is why current operation gets set right before sending a request to the broker.

Reply:
Oh well I did not know that! Thanks for clearing that up. I have removed that switch statement and the tests still pass so nothing was using that :)